src/server.ts

```diff
@@ -152,94 +118,50 @@ export class Server {
              *
              * Process Flow:
              * 1. Creates a new stream with a unique ID
-             * 2. Sets up storage for stream data and receivers
-             * 3. Forwards received chunks to all registered receivers
-             * 4. Cleans up resources when the stream ends
+             * 2. Notifies all instantiators that a stream message is pending
```

Comment:
Can't we determine beforehand which instantiator is of relevance here? We have the pipeline definition so we know which processor is consuming which reader, and we know which runner/instantiator is used for that processor.
This should suffice as we can only have one reader and thus one processor and thus one runner and thus one instatiator consuming the channel